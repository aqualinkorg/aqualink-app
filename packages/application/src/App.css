html, body { height: 100vh }
#root {
    display: flex;
    flex-direction: column;
    height: 100%;
}

#app {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
}

.MuiButtonBase-root.MuiIconButton-root.MuiPickersDay-day,
.MuiPickersCalendarHeader-switchHeader {
  color: black;
}
