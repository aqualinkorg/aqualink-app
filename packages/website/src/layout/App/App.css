html, body { height: 100vh }
#root {
    display: flex;
    flex-direction: column;
    height: 100%;
}

#app {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
}

.tooltip-arrow {
  width: 0px;
  height: 0px;
  border-style: solid;
  margin: auto;
  border-width: 10px 10px 0 10px;
  border-color: #404b6b transparent transparent transparent;
  opacity: 0.9;
  line-height: 0px;
}

.leaflet-popup-content-wrapper {
  border-radius: 4px;
  padding: 0px;
}

.leaflet-popup-content {
  border-radius: 4px;
  margin: 0px;
}

.leaflet-popup-tip {
  height: 12px;
  width: 12px;
}

.MuiInputBase-root.MuiOutlinedInput-root.MuiAutocomplete-inputRoot.MuiInputBase-fullWidth.MuiInputBase-formControl.MuiInputBase-adornedEnd.MuiOutlinedInput-adornedEnd {
  padding: 0;
}

.MuiAutocomplete-option {
  color: #168dbd;
}

.marker-icon {
  background-color: transparent;
}
