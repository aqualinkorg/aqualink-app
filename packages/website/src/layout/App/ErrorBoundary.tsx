/* eslint-disable react/prefer-stateless-function */
import React, { Component, PropsWithChildren } from "react";
import ErrorPage from "../../common/ErrorPage";

interface ErrorBoundaryState {
  hasError: boolean;
}

class ErrorBoundary extends Component<
  PropsWithChildren<{}>,
  ErrorBoundaryState
> {
  constructor(props: PropsWithChildren<{}>) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError = () => {
    return { hasError: true };
  };

  render() {
    const { hasError } = this.state;
    const { children } = this.props;

    return hasError ? <ErrorPage /> : children;
  }
}

export default ErrorBoundary;
