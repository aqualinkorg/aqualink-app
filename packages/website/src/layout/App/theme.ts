import { createMuiTheme } from "@material-ui/core/styles";

const skyBlue = "#009ee0";
const lightBlue = "#168dbd";
const lighterBlue = "#c0e1f0";
const darkGreyBlue = "#2d3436";
const black = "#2f2f2f";
const white = "#ffffff";
const lightGray = "#939393";

const fontFamily =
  "-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif";

export const colors = { skyBlue, lightBlue, lighterBlue, darkGreyBlue };

const theme: any = createMuiTheme({
  palette: {
    primary: {
      main: lightBlue,
      dark: darkGreyBlue,
      light: white,
    },
    text: {
      primary: white,
      secondary: black,
    },
    grey: {
      500: lightGray,
    },
  },
  overrides: {
    MuiAppBar: {
      root: {
        height: 122,
        justifyContent: "center",
        backgroundColor: lightBlue,
      },
    },
    MuiToolbar: {
      root: {
        justifyContent: "space-between",
      },
      dense: {
        minHeight: 40,
      },
    },
    MuiTypography: {
      h1: {
        fontSize: 52,
        fontFamily,
      },
      h2: {
        fontSize: 48,
        fontFamily,
      },
      h3: {
        fontSize: 32,
        fontFamily,
      },
      h4: {
        fontSize: 26,
        fontFamily,
      },
      h5: {
        fontSize: 20,
        fontFamily,
      },
      h6: {
        fontSize: 16,
        fontFamily,
      },
      subtitle1: {
        fontSize: 14,
        fontFamily,
      },
      subtitle2: {
        fontSize: 12,
        fontFamily,
      },
      caption: {
        fontSize: 10,
        fontFamily,
      },
      overline: {
        fontSize: 8.5,
        fontFamily,
      },
      gutterBottom: {
        marginBottom: "1rem",
      },
    },
    MuiGrid: {
      "spacing-xs-10": {
        width: "100%",
        margin: "0",
      },
    },
    MuiButton: {
      root: {
        borderRadius: 5,
      },
      containedPrimary: {
        backgroundColor: lightBlue,
      },
      containedSecondary: {
        backgroundColor: darkGreyBlue,
      },
    },
    MuiCardContent: {
      root: {
        "&:last-child": {
          paddingBottom: 0,
        },
      },
    },
    MuiInputLabel: {
      root: {
        color: lightGray,
      },
    },
    MuiInputBase: {
      root: {
        height: "100%",
      },
    },
    MuiOutlinedInput: {
      root: {
        padding: 0,
      },
    },
  },
});

export default theme;
