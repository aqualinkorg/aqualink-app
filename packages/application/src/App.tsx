import React from "react";
import { ThemeProvider } from "@material-ui/core/styles";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

import theme from "./theme";
import Form from "./Form";
import NotFound from "./NotFound";
import "./App.css";

function App() {
  return (
    <ThemeProvider theme={theme}>
      <Router>
        <div id="app">
          <Switch>
            {/* Redirect the home to Aqualink website */}
            <Route
              exact
              path="/"
              component={() => {
                window.location.replace("https://www.aqualink.org/");
                return null;
              }}
            />
            <Route exact path="/:appId/:uid" component={Form} />

            <Route default component={NotFound} />
          </Switch>
        </div>
      </Router>
    </ThemeProvider>
  );
}

export default App;
