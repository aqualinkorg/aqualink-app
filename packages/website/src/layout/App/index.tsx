import React, { useState, useEffect } from "react";
import { ThemeProvider } from "@material-ui/core/styles";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { useDispatch } from "react-redux";

import NotFound from "../../routes/NotFound";
import ErrorBoundary from "./ErrorBoundary";
import LandingPage from "../../routes/Landing";
import HomeMap from "../../routes/HomeMap";
import ReefRoutes from "../../routes/ReefRoutes";
import About from "../../routes/About";
import RegisterSite from "../../routes/RegisterSite";
import Buoy from "../../routes/Buoy";
import Drones from "../../routes/Drones";
import Faq from "../../routes/Faq";
import Dashboard from "../../routes/Dashboard";
import theme from "./theme";
import "leaflet/dist/leaflet.css";
import "./App.css";
import "../../assets/css/bootstrap.css";
import { getSelf } from "../../store/User/userSlice";
import app from "../../firebase";
import { initGA } from "../../utils/google-analytics";
import Terms from "../../routes/Terms";

function App() {
  const [render, setRender] = useState<boolean>(false);
  const dispatch = useDispatch();

  useEffect(() => {
    if (app) {
      app.auth().onAuthStateChanged((user) => {
        if (user) {
          // User is signed in
          user.getIdToken().then((token) => {
            dispatch(getSelf(token));
          });
        }
      });
    }
    setRender(true);
    initGA();
  }, [dispatch]);

  return (
    <ThemeProvider theme={theme}>
      <Router>
        <ErrorBoundary>
          <div id="app">
            {render && (
              <Switch>
                <Route exact path="/" component={LandingPage} />
                <Route exact path="/map" component={HomeMap} />
                <Route exact path="/about" component={About} />
                <Route exact path="/register" component={RegisterSite} />
                <Route exact path="/buoy" component={Buoy} />
                <Route exact path="/drones" component={Drones} />
                <Route exact path="/faq" component={Faq} />
                <Route exact path="/terms" component={Terms} />
                <Route path="/reefs" component={ReefRoutes} />
                <Route exact path="/dashboard" component={Dashboard} />
                <Route
                  exact
                  path="/collections/:collectionName"
                  component={Dashboard}
                />
                <Route default component={NotFound} />
              </Switch>
            )}
          </div>
        </ErrorBoundary>
      </Router>
    </ThemeProvider>
  );
}

export default App;
