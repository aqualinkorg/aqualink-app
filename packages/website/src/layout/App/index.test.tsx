import React from "react";
import { render } from "@testing-library/react";
import App from "./index";

jest.mock("../NavBar", () => "mock-navbar");

test("renders as expected", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
