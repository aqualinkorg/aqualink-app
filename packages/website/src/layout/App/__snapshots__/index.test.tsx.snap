// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-navbar />
  <div
    id="app"
  >
    <p>
      Route not found.
    </p>
  </div>
</div>
`;
